Gopkg.lock

```diff
@@ -1,332 +0,0 @@
-# This file is autogenerated, do not edit; changes may be undone by the next 'dep ensure'.
```

Comment:
Why is `Gopkg.lock` file deleted?

Reply:
Oh, that is a blunder. Apparently, `dep ensure` did not generate `Gopkg.lock` the file automatically if deleted. I was trying to update the dependencies as well as support for Golang v1.12.x. I had to make an empty lock file and then run `dep ensure`.

Reply:
I have re-commited changes, please check.

Reply:
There are some new dependencies added here, like `protobuf`. Are those required?

Reply:
Yes. It's a lock file, that means it is automatically generated. I have updated all the dependencies by deleting the lock file first, then initializing an empty one followed bt `dep ensure`